assembly/src/parsers/io_ops/mem_ops.rs

```diff
@@ -129,7 +129,8 @@ fn push_local_addr(
     // put the absolute memory address on the stack
     // negate the value to use it as an offset from the fmp
     // since the fmp value was incremented when locals were allocated
-    push_value(span_ops, -Felt::from(index));
+    // As the index increases, so does the memory address
+    push_value(span_ops, -Felt::from(num_proc_locals - 1 - index));
```

Comment:
I'd probably update the entire comment to something like: "put the absolute memory address on the stack; the absolute address is computed by subtracting index of the local from the `fmp` value. this way, the first local is located at `fmp - (num_proc_locals - 1)` (i.e., the smallest address) and the last local is located at `fmp` (i.e., the largest address)."

Also nit: would prefer `num_proc_locals - index - 1`.